src/usr/local/www/diag_testport.php

```diff
@@ -222,28 +222,28 @@
 if ($input_errors) {
 	print_input_errors($input_errors);
 } else {
-	// New page
-	if (empty($result) && $retval != 0 && !$showtext) {
-	    print('<div class="alert alert-warning" role="alert">This page allows you to perform a simple TCP connection test to determine if a host is up and accepting connections on a given port.' .
-	          ' This test does not function for UDP since there is no way to reliably determine if a UDP port accepts connections in this manner.</div>');
-	}
-
-	// Good host & port
-	if ($retval == 0 && $do_testport == 1)	{
-		if (!$showtext) {
-			print('<div class="alert alert-success" role="alert">'.gettext("Port test to host: " . $host . " Port: " . $port . " successful").'</div>');
-		} else {
-			print('<div class="alert alert-success" role="alert">'.gettext("Port test to host: " . $host . " Port: " . $port . " successful") . '. Any text received from the host will be shown below the form.</div>');
-		}
-	}
-
-	// netcat exit value != 0
-	if ($retval != 0 && !empty($result)) {
-		if ($showtext) {
-			print('<div class="alert alert-danger" role="alert">'.gettext('No output received, or connection failed. Try with "Show Remote Text" unchecked first.').'</div>');
+	if ($do_testport)	{
+		// User asked for a port test
+		if ($retval == 0)	{
+			// Good host & port
+			if (!$showtext) {
+				print('<div class="alert alert-success" role="alert">' . sprintf(gettext('Port test to host: %1$s Port: %2$s successful'), $host, $port) . '</div>');
+			} else {
+				print('<div class="alert alert-success" role="alert">' . sprintf(gettext('Port test to host: %1$s Port: %2$s successful.'), $host, $port) . ' ' . gettext('Any text received from the host will be shown below the form.') . '</div>');
```

Comment:
First sentence is the same in both cases, it could print it and then print 2nd sentence only when $showtext is true